core/src/main/java/org/elasticsearch/index/query/ConstantScoreQueryBuilder.java

```diff
@@ -66,20 +70,78 @@ public ConstantScoreQueryBuilder boost(float boost) {
         return this;
     }
 
+    /**
+     * @return the boost factor
+     */
+    public float boost() {
+        return this.boost;
+    }
+
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(NAME);
         builder.field("filter");
-        filterBuilder.toXContent(builder, params);
+        queryBuilder.toXContent(builder, params);
+        builder.field("boost", boost);
+        builder.endObject();
+    }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        Query innerFilter = queryBuilder.toQuery(parseContext);
+        if (innerFilter == null ) {
+            // no inner query, we ignore this and return no docs
+            return Queries.newMatchNoDocsQuery();
```

Comment:
know what? I can't make up my mind here :)  it is weird to have checks for things that won't ever happen cause we do rely on validation...on the other hand toQuery is public and can be called from anywhere (maybe we can change that one day though) in which case we have to watch out for all the NPEs and weird things that can happen. Do you have a preference?